### PR TITLE
Calculator

### PR DESCRIPTION
**Pull request by phillipwright7**
## Summary

I've created a calculator executable and a test file to accompany it.

## Description

The `calculator` executable performs the four math operators, and the test file `calulcator_test` will utilize table-driven tests and subtests to make sure the expected results are correct.

### CODE REVIEW

File: cmd/calculator/calculator.go [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+func Multiply(a, b int) int {
+	pro := a * b
+	return pro
+}
+
+func Divide(a, b int) (int, error) {
+	if b == 0 {
+		return 0, errors.New("divide by zero error")
+	} else {
```
Comment (jboursiquot) [inline cmd/calculator/calculator.go:23]: The `else` statement is not necessary here.

File: cmd/calculator/calculator.go [hunk 2]
```diff
@@ -0,0 +1,27 @@
+package calculator
+
+import "errors"
+
+func Add(a, b int) int {
+	sum := a + b
```
Comment (jboursiquot) [inline cmd/calculator/calculator.go:6]: You can omit the variable assignment, it's not used anywhere. Just return the result of the operation.

File: cmd/calculator/calculator_test.go [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+func TestDivide(t *testing.T) {
+	tests := []struct {
+		name                string
+		input, input2, want int
+	}{
+		{name: "first", input: 2, input2: 2, want: 0},
+		{name: "second", input: 2, input2: 3, want: 1},
+		{name: "third", input: 0, input2: 2, want: 0},
+		{name: "fourth", input: -1, input2: 2, want: 0},
```
Comment (jboursiquot) [inline cmd/calculator/calculator_test.go:77]: How might you test for the divide-by-zero error?

**Review by jboursiquot — APPROVED** [review]
Nice work. Address the minor tweaks and you're good to go.

File: cmd/calculator/calculator_test.go (hunk 3 above)
Comment (phillipwright7) [reply cmd/calculator/calculator_test.go:77]: Why might my terminal format the `%q` verbs in this weird fashion?

image

**Pull request merged**